pontoon/actionlog/migrations/0004_auto_20240816_1045.py

```diff
@@ -0,0 +1,81 @@
+# Generated by Django 4.2.11 on 2024-08-16 10:45
+
+import datetime
+
+from django.db import migrations
+
+
+def migrate_translation_to_actionlog(apps, schema_editor):
+    Translation = apps.get_model("base", "Translation")
+    ActionLog = apps.get_model("actionlog", "ActionLog")
+
+    translation_info = {
+        "date": (("translation:created", "Translation created"), "user"),
+        "approved_date": (
+            ("translation:approved", "Translation approved"),
+            "approved_user",
+        ),
+        "unapproved_date": (
+            ("translation:unapproved", "Translation unapproved"),
+            "unapproved_user",
+        ),
+        "rejected_date": (
+            ("translation:rejected", "Translation rejected"),
+            "rejected_user",
+        ),
+        "unrejected_date": (
+            ("translation:unrejected", "Translation unrejected"),
+            "unrejected_user",
+        ),
+    }
+
+    # date to end migration
+    end_date = datetime.datetime(
+        2020, 1, 7, 9, 25, 11, 829125, tzinfo=datetime.timezone.utc
+    )
+
+    actions_to_log = []
+
+    # Exclude translations that don't have any user set to improve performance
+    translations = Translation.objects.filter(date__lt=end_date).exclude(
+        user=None,
+        approved_user=None,
+        unapproved_user=None,
+        rejected_user=None,
+        unrejected_user=None,
+    )
+
+    for translation in translations:
+        for attr, (action_type, action_user) in translation_info.items():
+            date = getattr(translation, attr)
+            user_id = getattr(translation, action_user)
+
+            # Skip logging 'translation:approved' if approved_date is the same as the date
```

Comment:
Thank you for the feedback. I agree that comments should be more focused on explaining the reasoning behind the code rather than just describing its functionality.

I’ll update the comments to clarify why we handle logging in this way. It should make the code’s intent clearer and provide better context for future readers.

Thanks again for pointing this out!